### ISSUE TITLE
android.os.NetworkOnMainThreadException

### ISSUE THREAD

**Issue by Hakog** [MobileChromeApps/cordova-plugin-chrome-apps-sockets-tcp#12]
Hi,

An `android.os.NetworkOnMainThreadException` is thrown when trying to connect. This exception is thrown only at Android 4.1.1 API level 16. Everything above Android 4.1.1 API level 16 seems to work fine. 

Tested with version 1.3.4 of the plugin.

Should the connect() method run on a seperate thread ? Or is this an issue on Android 4.1.1 only ? Or maybe i am doing something wrong ?

I am using the plugin inside an ionic project.

Something like:
```
    window.chrome.sockets.tcp.create({}, function(createInfo) {
      console.log('socket created ', JSON.stringify(createInfo));
      window.chrome.sockets.tcp.connect(createInfo.socketId, '127.0.0.1', 8100, function(connectionInfo) {
        console.log('socket connected ', JSON.stringify(connectionInfo));
        var message = str2ab('hello world');
        window.chrome.sockets.tcp.send(createInfo.socketId, message, function(sendInfo) {
          console.log('Message send ', JSON.stringify(sendInfo));
        });
      });
    });
```
Complete stacktrace:
```

E/PluginManager( 1228): android.os.NetworkOnMainThreadException
E/PluginManager( 1228): 	at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1117)
E/PluginManager( 1228): 	at libcore.io.BlockGuardOs.connect(BlockGuardOs.java:84)
E/PluginManager( 1228): 	at libcore.io.IoBridge.connectErrno(IoBridge.java:127)
E/PluginManager( 1228): 	at libcore.io.IoBridge.connect(IoBridge.java:112)
E/PluginManager( 1228): 	at libcore.io.IoBridge.connect(IoBridge.java:100)
E/PluginManager( 1228): 	at java.nio.SocketChannelImpl.connect(SocketChannelImpl.java:168)
E/PluginManager( 1228): 	at org.chromium.ChromeSocketsTcp$TcpSocket.connect(ChromeSocketsTcp.java:779)
E/PluginManager( 1228): 	at org.chromium.ChromeSocketsTcp.connect(ChromeSocketsTcp.java:239)
E/PluginManager( 1228): 	at org.chromium.ChromeSocketsTcp.execute(ChromeSocketsTcp.java:63)
E/PluginManager( 1228): 	at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:117)
E/PluginManager( 1228): 	at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98)
E/PluginManager( 1228): 	at org.apache.cordova.PluginManager.exec(PluginManager.java:133)
E/PluginManager( 1228): 	at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59)
E/PluginManager( 1228): 	at org.apache.cordova.CordovaBridge.promptOnJsPrompt(CordovaBridge.java:135)
E/PluginManager( 1228): 	at org.apache.cordova.engine.SystemWebChromeClient.onJsPrompt(SystemWebChromeClient.java:124)
E/PluginManager( 1228): 	at android.webkit.CallbackProxy.handleMessage(CallbackProxy.java:655)
E/PluginManager( 1228): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/PluginManager( 1228): 	at android.os.Looper.loop(Looper.java:137)
E/PluginManager( 1228): 	at android.app.ActivityThread.main(ActivityThread.java:4745)
E/PluginManager( 1228): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/PluginManager( 1228): 	at java.lang.reflect.Method.invoke(Method.java:511)
E/PluginManager( 1228): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
E/PluginManager( 1228): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
E/PluginManager( 1228): 	at dalvik.system.NativeStart.main(Native Method)
```

**Comment by bemasc** [comment]
Possibly related: on a Nexus 5X with Android M, API level 24, I'm seeing messages like
`W/PluginManager: THREAD WARNING: exec() call to ChromeSocketsTcp.connect blocked the main thread for 458ms. Plugin should use CordovaInterface.getThreadPool().`

458 ms seems like a long time.  Maybe this is a blocking API that shouldn't be used on this thread?

**Comment by profOnno** [comment]
I have the same problem on Android 4.1.2
```
E/PluginManager(10586): Uncaught exception from plugin
E/PluginManager(10586): android.os.NetworkOnMainThreadException
E/PluginManager(10586): 	at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1117)
E/PluginManager(10586): 	at libcore.io.BlockGuardOs.connect(BlockGuardOs.java:84)
E/PluginManager(10586): 	at libcore.io.IoBridge.connectErrno(IoBridge.java:127)
....
```

The older `cordova-plugin-chrome-apps-socket` works on android 4.1.2.
But that version of `ChromeSocket.java` wraps connect in a ConnectThread class. Was there a point of doing this different in the newer `cordova-plugin-chrome-apps-sockets-tcp` ?
Or is this just the 'new' android way, and there was no basis for keeping compatible with the older android versions?

**Comment by agrieve** [comment]
Certainly looks like `.connect()` should be run on a dedicated background thread.

**Comment by agrieve** [comment]
Published in v1.3.6
